src/main/java/be/seeseemelk/mockbukkit/block/data/BlockDataMock.java

```diff
@@ -201,23 +211,65 @@ public boolean equals(Object obj)
 	{
 		Preconditions.checkNotNull(material, "Material cannot be null");
 		// Special Cases
+		BlockDataMock mock = attemptMockByPaperMaterialTags(material);
+		if (mock != null)
+		{
+			return mock;
+		}
+
+		mock = attemptMockByTag(material);
+		if (mock != null)
+		{
+			return mock;
+		}
+
+		return switch (material)
+				{
+					case AMETHYST_CLUSTER -> new AmethystClusterMock(material);
+					default -> new BlockDataMock(material);
+				};
+	}
+
+	/**
+	 * Attempts to construct a BlockDataMock object by matching against Paper MaterialTags. Returns null if the given
+	 * material does not match any supported MaterialSetTag.
+	 *
+	 * @param material Material which we will attempt to mock
+	 * @return BlockDataMock if matched, null otherwise
+	 */
+	static BlockDataMock attemptMockByPaperMaterialTags(@NotNull Material material)
+	{
+		Preconditions.checkNotNull(material, "Material cannot be null");
 		if (MaterialTags.BEDS.isTagged(material))
 		{
 			return new BedMock(material);
 		}
-		else if (Tag.SLABS.isTagged(material))
+		return null;
+	}
+
+	/**
+	 * Attempts to construct a BlockDataMock object by matching against Bukkit Tags. Returns null if the given material
+	 * does not match any supported Tag.
+	 *
+	 * @param material Material which we will attempt to mock
+	 * @return BlockDataMock if matched, null otherwise
+	 */
+	static BlockDataMock attemptMockByTag(@NotNull Material material)
```

Comment:
Can this be private? I don't see a need for anything to ever access this outside of the mock method.